plugins/mumble_plugin_utils.h

```diff
@@ -49,4 +54,43 @@ static void escape(char *str, size_t size) {
     }
 }
 
+// Set a string securely.
+static inline size_t StrSet(char *str, size_t size, const char *fmt, ...) {
+	va_list args;
+
+	va_start(args, fmt);
+	int len = vsnprintf(str, size, fmt, args);
+	va_end(args);
+
+	return len;
+}
+
+// Copy a string securely.
+static inline size_t StrCpy(char *dst, size_t size, const char *src) {
+	// Validate arguments
+	if (dst == NULL || src == NULL || dst == src || size == 0) {
+		return 0;
+	}
+
+	// Append NUL character and return if there's space only for it
+	if (size == 1) {
+		dst[0] = '\0';
+		return 0;
+	}
+
+	// Get source string length and decrease it if necessary
+	size_t len = strnlen(src, size);
+	if (len >= size) {
+		len = size - 1;
```

Comment:
The method comment and maybe also this block comment should document that if size is smaller than src that only size - 1 characters will be copied.

Reply:
```
// This function copies the number of chars specified.
// If "size" is smaller than "src", the number of chars to be copied corresponds to "size" - 1, as the last space available is required to NUL-terminate the string.
```

Reply:
Hi, I don't know where can I make this question but I'll try here.
I'm new on Mumble but if I want to make changes in this service application is possible to make in an IDE running on Windows?
What IDE you're using to run mumble to debug?
I'm doing by this way, changin a file and then build the application after that, deploing in a Ubuntu server to test it. For me to do this by this way is very hard and I think there is an easy  way to do this. can you help me?

Reply:
Hi, I personally use Qt Creator on both Linux and Windows, because it's deeply integrated with Qt.